scanners/zap/zap.py

```diff
@@ -270,15 +270,17 @@ def _setup_zap_automation(self):
         # Configure the basic environment target
         try:
             af_context = find_context(self.automation_config)
-            af_context["urls"].append(self.config.get("application.url"))
+            app_url = self.config.get("application.url")
+            if app_url:
```

Comment:
RE: "The change I propose is to not add an empty URL.  ZAP will still refuse to start with a faulty config, but at least the shutdown is graceful and there is a clear error message.", why don't we just exit here giving the user an error to set the application.url?

Reply:
Good point. updated